### ISSUE TITLE
ndarray mmul on two vectors returns nil

### ISSUE THREAD

**Issue by mars0i** [mikera/core.matrix#145]
When 1D vectors are passed to `mmul`, the intended behavior is that the first one is treated as a row vector, and the second one is treated as a column vector, so that multiplying a 1D vector by a 1D vector returns their inner product (see [wiki page](https://github.com/mikera/core.matrix/wiki/Vectors-vs.-matrices)).  This works in the persistent-vector and vectorz-clj implementations, but currently returns `nil` in the ndarray implementation.  (The function `inner-product` produces the correct result in all implementations.)

In persistent-vector and vectorz-clj, `mmul` does the right thing: 

``` clojure
user=> (inner-product (matrix :persistent-vector [1 2 3]) (matrix :persistent-vector [1 2 3]))
14
user=> (mmul (matrix :persistent-vector [1 2 3]) (matrix :persistent-vector [1 2 3]))
14
user=> (inner-product (matrix :vectorz [1 2 3]) (matrix :vectorz [1 2 3]))
#<Scalar 14.0>
user=> (mmul (matrix :vectorz [1 2 3]) (matrix :vectorz [1 2 3]))
#<Scalar 14.0>
```

(I assume the wrapped number is what's intended in vectorz-clj.)
In ndarray `persistent-vector` behaves correctly, but `mmul` on two 1D vectors returns nil:

``` clojure
user=> (inner-product (matrix :ndarray [1 2 3]) (matrix :ndarray [1 2 3]))
14
user=> (mmul (matrix :ndarray [1 2 3]) (matrix :ndarray [1 2 3]))
nil
```

Note that ndarray's implementation of `mmul` behaves correctly on true row and column matrices, and on 1D vectors when the other argument is 2D:

``` clojure
user=> (mmul (matrix :ndarray [[1 2 3]]) (matrix :ndarray [[1] [2] [3]]))
#<NDArray [[14.0]]>
user=> (mmul (matrix :ndarray [[1 2 3][1 2 3][1 2 3]]) (matrix :ndarray [1 2 3]))
#<NDArray [14.0 14.0 14.0]>
user=> (mmul (matrix :ndarray [1 2 3]) (matrix :ndarray [[1 2 3][1 2 3][1 2 3]])))
#<NDArray [6.0 12.0 18.0]>
```

**Comment by mars0i** [comment]
The problem is in `clojure.core.matrix.impl.ndarray/matrix-multiply`.  The vector X vector case is falling through the long `cond` expression because there's no case for a-ndims = 1, b-ndims =1, i.e. no case for the situation in which both arguments to `mmul` have 1 dimension.  Fixed it, submitting a pull request.  Added a test for this case to compliance_tester.clj.  (Seemed like the best place; I did didn't see tests for other mmul cases, but I might have missed it.)  See what you think.
